IPython/core/oinspect.py

```diff
@@ -630,10 +631,10 @@ def code_formatter(text):
             # Functions, methods, classes
             append_field(_mime, 'Signature', 'definition', code_formatter)
             append_field(_mime, 'Init signature', 'init_definition', code_formatter)
+            append_field(_mime, 'Docstring', 'docstring', formatter)
             if detail_level > 0 and info['source']:
                 append_field(_mime, 'Source', 'source', code_formatter)
             else:
-                append_field(_mime, 'Docstring', 'docstring', formatter)
                 append_field(_mime, 'Init docstring', 'init_docstring', formatter)
```

Comment:
I wonder if a similar argument can be used for moving this to become unconditional (although it's perhaps out of scope for this change)

Reply:
Again, this would mostly make sense with a proper refactoring so that the same AST test can be applied to the init docstring, but that'll have to go in another PR.